sunpy/visualization/imageanimator.py

```diff
@@ -608,32 +585,217 @@ def _sanitize_axis_range(self, axis_range, data):
+        extent = []
+        # reverse because numpy is in y-x and extent is x-y
+        for i in self.image_axes[::-1]:
+            extent += self.axis_range[i]
+
+        imshow_args = {'interpolation': 'nearest',
+                       'origin': 'lower',
+                       'extent': extent,
+                       }
+
+        imshow_args.update(self.imshow_kwargs)
+        im = ax.imshow(self.data[self.frame_slice], **imshow_args)
+        if self.if_colorbar:
+            self._add_colorbar(im)
+
+        return im
+
+    def update_plot(self, val, im, slider):
         val = int(val)
-        ax = self.slider_axes[slider.slider_ind]
-        ind = np.argmin(np.abs(self.axis_range[ax] - val))
-        self.frame_slice[ax] = ind
+        ax_ind = self.slider_axes[slider.slider_ind]
+        ind = np.argmin(np.abs(self.axis_range[ax_ind] - val))
+        self.frame_slice[ax_ind] = ind
         if val != slider.cval:
             im.set_array(self.data[self.frame_slice])
             slider.cval = val
+
+
+class LineAnimator(ArrayAnimator):
+    """Create a matplotlib backend independent data explorer for 1D plots.
+
+    The following keyboard shortcuts are defined in the viewer:
+
+    - 'left': previous step on active slider
+    - 'right': next step on active slider
+    - 'top': change the active slider up one
+    - 'bottom': change the active slider down one
+    - 'p': play/pause active slider
+
+    This viewer can have user defined buttons added by specifying the labels
+    and functions called when those buttons are clicked as keyword arguments.
+
+    Parameters
+    ----------
+    ydata: ndarray
+        The y-axis data to be visualised = 2D
+
+    xdata: ndarray
+        The x-axis data against which the slices of the ydata is plotted.
+        Must 1D of same length as the image_axis of ydata.
+        If None, array indices are used.
+
+    image_axis: `int`
+        The axis used to plot against xdata.
+
+    fig: `matplotlib.figure`
+        Figure to use
+
+    axis_range: list of physical coordinates for array or None
+        If None array indices will be used for all axes.
+        If a list it should contain one element for each axis of the numpy array.
+        For the image axes a [min, max] pair should be specified which will be
+        passed to :func:`matplotlib.pyplot.plot` as extent.
+        For the slider axes a [min, max] pair can be specified or an array the
+        same length as the axis which will provide all values for that slider.
+        If None is specified for an axis then the array indices will be used
+        for that axis.
+
+    interval: int
+        Animation interval in ms
+
+    button_labels: list
+        List of strings to label buttons
+
+    button_func: list
+        List of functions to map to the buttons
+
+    Extra keywords are passed to plot.
+
+    """
+
+    def __init__(self, ydata, xdata=None, image_axis=0, axis_range=None, xlim=None, ylim=None,
```

Comment:
But because axis_range expects lengths/dimensions useful for 2D and greater, it requires a very non-intuitive way for a user to enter the info when thinking about 1D data.  I think it's better to derive axis_range from xdata rather than the other way around in the case of LineAnimator.